### ISSUE TITLE
BUG: unable to match curly brackets

### ISSUE THREAD

**Issue by GnoStiC** [facelessuser/BracketHighlighter#219]
*BUG*: for function's Curly shows only **?**
int foo(a, b) {
	MYFUNC ("$meta)");
}

*This works OK:*
int foo() {
	DEBUG (" $f ");
}

*This also works OK:*
int foo() {
	_DEBUG (" $f) ")
}

**Comment by facelessuser** [comment]
I am not experiencing this. 

![screenshot - 1_29_2015 8_55_01 am](https://cloud.githubusercontent.com/assets/1055125/5960718/ba0086f6-a794-11e4-8681-aac41b1e6554.png)

I am assuming you are using a C/C++ syntax highlighter?  I am using a custom syntax highlighter, but it should be noted that if a syntax highlighter is not highlighting things appropriately, it can have an effect.

Anyways, please be more specific about the details of the environment: what syntax highlighter are you using, ST2 or ST3, etc.

But I will say this, there doesn't appear to inherently be a problem with the algorithm, if there is a problem, I suspect it is with the code view's environment not playing nice with the current settings and/or bracket rules.

**Comment by GnoStiC** [comment]
* ST3, build 3065 (registered)
* Syntax = Groovy (and you're right, if i use C syntax everything is as it should be)

![screen shot 2015-01-29 at 18 27 48](https://cloud.githubusercontent.com/assets/140036/5961334/af1fe240-a7e4-11e4-9417-f7f9de181551.jpg)

**Comment by facelessuser** [comment]
Ahh, groovy; I see.  I don't use groovy, but after quickly switching to the syntax I see some problems with the groovy syntax.  Using another of my plugins [Scope Hunter](https://github.com/facelessuser/ScopeHunter)  I was able to conclude that on line `2`, that the first parenthesis is scoped as the opening, the second (right after meta) is scoped as the end, and the actual end is not scoped as a end bracket at all.

Now, the round bracket rule doesn't really care about the scope except to make sure a round bracket is not a string or comment...but the groovy syntax doesn't seem to scope strings as strings.  So the default groovy syntax kind of sucks.

Basically, since the bracket right after meta is not scoped as part of the string, BH thinks that all three round brackets an that line are valid.  So now we have a condition like this: `{ ( ) ) }`.  BH can't resolve the second closing round bracket because the first is getting paired with the opening throwing off the matching of the curly.

So I will be honest, what you need is a better groovy highlighter.

**Comment by facelessuser** [comment]
Correction.  The groovy syntax highlighter doesn't seem to scope the first or second string as double quoted strings, but it does scope the last string proper which is why you don't see the same problem on the last function.

**Comment by facelessuser** [comment]
As this is not an issue with BH, this issue will be closed.
